src/index.js

```diff
@@ -0,0 +1,113 @@
+const MAX_TIMEOUT_ATTEMPTS = 3;
+const MAX_TIMEOUT = 1000;
+
+
+function main(client) {
```

Comment:
I would also include a preflight check that `Promise` exists. We assume that `Promise` is supported or has been polyfilled.